crates/starknet-devnet/tests/test_simulate_transactions.rs

```diff
@@ -232,6 +234,204 @@ mod simulation_tests {
+        let get_params = |simulation_flags: &[&str]| -> serde_json::Value {
+            json!({
+                "block_id": "latest",
+                "simulation_flags": simulation_flags,
+                "transactions": [
+                    {
+                        "type": "DECLARE",
+                        "sender_address": sender_address_hex,
+                        "compiled_class_hash": to_hex_felt(&casm_hash),
+                        "max_fee": to_hex_felt(&max_fee),
+                        "version": "0x2",
+                        "signature": signature_hex,
+                        "nonce": to_num_as_hex(&nonce),
+                        "contract_class": flattened_contract_artifact,
+                    }
+                ]
+            })
+        };
+
+        let chain_id = &devnet.send_custom_rpc("starknet_chainId", json!({})).await.unwrap();
+        assert_eq!(chain_id, assert_chain_id);
+
+        let params_no_flags = get_params(&[]);
+        let resp_no_flags = &devnet
+            .send_custom_rpc("starknet_simulateTransactions", params_no_flags.clone())
+            .await
+            .unwrap()[0];
+        assert_eq!(resp_no_flags["fee_estimation"]["gas_price"], "0x174876e800");
+        assert_eq!(resp_no_flags["fee_estimation"]["data_gas_price"], "0x174876e800");
+        assert_eq!(resp_no_flags["fee_estimation"]["overall_fee"], "0x7352ecf52000");
+
+        let params_skip_validation = get_params(&["SKIP_VALIDATE"]);
+        let resp_skip_validation = &devnet
+            .send_custom_rpc("starknet_simulateTransactions", params_skip_validation.clone())
+            .await
+            .unwrap()[0];
+        assert_eq!(resp_skip_validation["fee_estimation"]["gas_price"], "0x174876e800");
+        assert_eq!(resp_skip_validation["fee_estimation"]["data_gas_price"], "0x174876e800");
+        assert_eq!(resp_skip_validation["fee_estimation"]["overall_fee"], "0x733ba47e3800");
+
+        assert_difference_if_validation(
+            resp_no_flags,
+            resp_skip_validation,
+            &sender_address_hex,
+            max_fee == FieldElement::ZERO,
+        );
+
+        let gas_update = json!({
+            "gas_price_wei": 9 * 1e18 as u128,
+            "data_gas_price_wei": 8 * 1e18 as u128,
+            "gas_price_strk": 7 * 1e18 as u128,
+            "data_gas_price_strk": 6 * 1e18 as u128,
+            "generate_block": false,
+        });
+        let updated_gas =
+            &devnet.send_custom_rpc("devnet_updateGas", gas_update.clone()).await.unwrap();
+        assert_eq!(updated_gas, &gas_update);
+
+        let chain_id = &devnet.send_custom_rpc("starknet_chainId", json!({})).await.unwrap();
+        assert_eq!(chain_id, assert_chain_id);
+
+        let resp_no_flags = &devnet
+            .send_custom_rpc("starknet_simulateTransactions", params_no_flags)
+            .await
+            .unwrap()[0];
+        assert_eq!(resp_no_flags["fee_estimation"]["gas_price"], "0x7ce66c50e2840000");
```

Comment:
Here and in other places, what are all these hardcoded hex strings, can't they be calculated from something?

Reply:
Fixed in 6e4de95f65f2287f6fdc371b4af70fc8988acb5f

Reply:
these are not magic numbers, yes these gas fees can be calculated/predicted but the effort to do so is enormous  and it's just easier to change tests once the gas fee calculation mechanism will change

Reply:
reverted to hexex in 1c04bcb76f3aa9af05a748ecb2c71940aee256e7